Formula/samba.rb

```diff
@@ -25,6 +25,7 @@ class Samba < Formula
   # configure requires python3 binary to be present, even when --disable-python is set.
   depends_on "python@3.10" => :build
   depends_on "gnutls"
+  depends_on "krb5"
```

Comment:
Minimum version is 1.19, so cannot use macOS system `krb5`.

Depending on `heimdal` also has some build issues. Not sure if there is any major behavior difference between MIT `krb5` and Heimdal, or just licensing.